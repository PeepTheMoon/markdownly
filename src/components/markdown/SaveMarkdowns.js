import React, { useState } from 'react';
import { useSelector } from '../../hooks/Context';
import { getMarkdown } from '../../selectors/selectors';

const SaveMarkdown = () => {
  const [title, setTitle] = useState('');
  const markdown = useSelector(getMarkdown);

  const handleChange = ({ target }) => {
    setTitle(target.value);
  };

  const handleSubmit = event => {
    event.preventDefault();

    
    dispatch(saveMarkdown({
      title: title, 
      markdown: markdown
    }));

  };
  
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" placeholder="Title" onChange={handleChange} value={title}/>
      <button>Save</button>
    </form>
  );
};

export default SaveMarkdown;
