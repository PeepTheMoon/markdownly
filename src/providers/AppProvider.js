import React, { useReducer } from 'react';
import { Context } from '../hooks/Context';
import reducer, { initialState } from '../reducer/AppReducer';

const AppProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <Context.Provider value={{ state, dispatch }}>
      { children }
    </Context.Provider>
  );
};

export default AppProvider;
