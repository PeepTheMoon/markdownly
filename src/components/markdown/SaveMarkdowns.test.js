import React from 'react';
import { mount } from 'enzyme';
import SaveMarkdowns from './SaveMarkdowns';
import AppProvider from '../../providers/AppProvider';

describe('SaveMarkdowns component', () => {
  it('renders SaveMarkdowns', () => {
    const wrapper = mount(
      <AppProvider>
        <SaveMarkdowns/>
      </AppProvider> 
    );
    expect(wrapper).toMatchSnapshot();
  });
});
