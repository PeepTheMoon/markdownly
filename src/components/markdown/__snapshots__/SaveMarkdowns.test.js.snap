// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveMarkdowns component renders SaveMarkdowns 1`] = `
<AppProvider>
  <SaveMarkdown>
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="Title"
        type="text"
        value=""
      />
      <button>
        Save
      </button>
    </form>
  </SaveMarkdown>
</AppProvider>
`;
